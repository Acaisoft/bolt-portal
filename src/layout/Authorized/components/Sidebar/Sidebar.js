import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Link, withRouter } from 'react-router-dom'
import {
  Avatar,
  Drawer,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Typography,
  withStyles,
} from '@material-ui/core'
import { Dashboard, Extension, ExitToApp, ShowChart } from '@material-ui/icons'

import styles from './Sidebar.styles'

export class Sidebar extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
  }

  items = [
    { label: 'Dashboard', linkTo: '/dashboard', icon: <Dashboard /> },
    { label: 'Projects', linkTo: '/projects', icon: <Extension /> },
    { label: 'Tests Results', linkTo: '/tests-results', icon: <ShowChart /> },
  ]

  render() {
    const { classes } = this.props

    return (
      <Drawer
        className={classes.drawer}
        variant="permanent"
        classes={{
          paper: classes.drawerPaper,
        }}
        anchor="left"
      >
        <div className={classes.text}>
          <div className={classes.intro}>
            <Avatar>A</Avatar>
            <Typography
              variant="subtitle1"
              color="inherit"
              gutterBottom
              className={classes.introText}
            >
              HELLO, Admin!
            </Typography>
          </div>
          <p>Another great day, for testing your apps!</p>
        </div>
        <List>
          {this.items.map(({ label, linkTo, icon }) => (
            <ListItem
              button
              key={linkTo}
              component={Link}
              to={linkTo}
              color="inherit"
              className={classes.listItem}
            >
              <ListItemIcon className={classes.itemIcon}>{icon}</ListItemIcon>
              <ListItemText
                disableTypography
                primary={
                  <Typography
                    type="body2"
                    style={{
                      color: '#E8E8E8',
                      fontSize: '1.25rem',
                      textTransform: 'uppercase',
                      fontWeight: 'normal',
                    }}
                  >
                    {label}
                  </Typography>
                }
              />
            </ListItem>
          ))}
        </List>
        <List>
          <ListItem button color="inherit" className={classes.listItem}>
            <ListItemIcon className={classes.itemIcon}>
              <ExitToApp />
            </ListItemIcon>
            <ListItemText
              disableTypography
              primary={
                <Typography
                  type="body2"
                  style={{
                    color: '#E8E8E8',
                    fontSize: '1.25rem',
                    textTransform: 'uppercase',
                    fontWeight: 'normal',
                  }}
                >
                  log out
                </Typography>
              }
            />
          </ListItem>
        </List>
      </Drawer>
    )
  }
}

export default withRouter(withStyles(styles)(Sidebar))
