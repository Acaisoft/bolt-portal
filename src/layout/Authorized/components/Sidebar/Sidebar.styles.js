const drawerWidth = 260

export default ({ palette, spacing }) => ({
  root: {
    display: 'flex',
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
    color: palette.gray.default,
  },
  drawerPaper: {
    width: drawerWidth,
    backgroundColor: palette.primary.main,
    padding: `${spacing.unit * 3}px ${spacing.unit * 2}px`,
  },
  text: {
    color: palette.gray.default,
  },
  intro: {
    display: 'flex',
    alignItems: 'center',
  },
  introText: {
    margin: 0,
    marginLeft: 15,
  },
  itemIcon: {
    color: palette.gray.default,
    marginRight: 0,
  },
  listItem: {
    paddingLeft: 0,
    paddingRight: 0,
  },
})
