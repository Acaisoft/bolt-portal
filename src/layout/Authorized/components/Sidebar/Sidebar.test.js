import React from 'react'
import { shallow } from 'enzyme'

import { Sidebar } from './Sidebar'
import { mocks } from '~utils/tests'

const initSidebar = overrides => {
  const wrapper = shallow(<Sidebar classes={mocks.ClassesProxy} {...overrides} />)
  return { wrapper }
}

describe('layout: Sidebar', () => {
  describe('rendering', () => {
    it('should render without crashing', () => {
      const { wrapper } = initSidebar()
      expect(wrapper).toBeTruthy()
    })

    it('should render properly', () => {
      const { wrapper } = initSidebar()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
