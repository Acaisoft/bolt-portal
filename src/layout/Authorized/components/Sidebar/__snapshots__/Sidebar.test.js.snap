// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout: Sidebar rendering should render properly 1`] = `
<DrawerMock
  anchor="left"
  className="drawer"
  classes={
    Object {
      "paper": "drawerPaper",
    }
  }
  variant="permanent"
>
  <div
    className="text"
  >
    <div
      className="intro"
    >
      <AvatarMock>
        A
      </AvatarMock>
      <TypographyMock
        className="introText"
        color="inherit"
        gutterBottom={true}
        variant="subtitle1"
      >
        HELLO, Admin!
      </TypographyMock>
    </div>
    <p>
      Another great day, for testing your apps!
    </p>
  </div>
  <ListMock>
    <ListItemMock
      button={true}
      className="listItem"
      color="inherit"
      component="LinkMock"
      key="/projects"
      to="/projects"
    >
      <ListItemIconMock
        className="itemIcon"
      >
        <ExtensionMock />
      </ListItemIconMock>
      <ListItemTextMock
        disableTypography={true}
        primary={
          <TypographyMock
            style={
              Object {
                "color": "#E8E8E8",
                "fontSize": "1.25rem",
                "fontWeight": "normal",
                "textTransform": "uppercase",
              }
            }
            type="body2"
          >
            Projects
          </TypographyMock>
        }
      />
    </ListItemMock>
    <ListItemMock
      button={true}
      className="listItem"
      color="inherit"
      component="LinkMock"
      key="/test-runs"
      to="/test-runs"
    >
      <ListItemIconMock
        className="itemIcon"
      >
        <ShowChartMock />
      </ListItemIconMock>
      <ListItemTextMock
        disableTypography={true}
        primary={
          <TypographyMock
            style={
              Object {
                "color": "#E8E8E8",
                "fontSize": "1.25rem",
                "fontWeight": "normal",
                "textTransform": "uppercase",
              }
            }
            type="body2"
          >
            Tests Results
          </TypographyMock>
        }
      />
    </ListItemMock>
  </ListMock>
  <ListMock>
    <ListItemMock
      button={true}
      className="listItem"
      color="inherit"
    >
      <ListItemIconMock
        className="itemIcon"
      >
        <ExitToAppMock />
      </ListItemIconMock>
      <ListItemTextMock
        disableTypography={true}
        primary={
          <TypographyMock
            style={
              Object {
                "color": "#E8E8E8",
                "fontSize": "1.25rem",
                "fontWeight": "normal",
                "textTransform": "uppercase",
              }
            }
            type="body2"
          >
            log out
          </TypographyMock>
        }
      />
    </ListItemMock>
  </ListMock>
</DrawerMock>
`;
